### PR TITLE
Adham data base homework week 3'

### PR DESCRIPTION
**Pull request by Adham-Elias-Botours**
##### Thanks

### CODE REVIEW

**Comment by Adham-Elias-Botours** [comment]
### Somthing is worng
*I am sure that before writing any code I had checked out to the master branch then I  checked out to new branch (week 3).*
**But now I see that is not true, so I have done something wrong.**
*so excuse me.*

File: Week3/homework/app.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+const tasks_routes = require('./api/routes/tasks_routes');
+
+app.use(express.json());
+
+app.all('/', (req, res, next) => {
+    res.status(201).json({
+        message: 'welcome to ToDo lists app',
+    });
+});
+app.use('/lists', lists_routes);
```
Comment (roxana-lafuente) [inline Week3/homework/app.js]: Super good!

File: Week3/homework/readme.md [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+**Insert the folowing in PostMan then send the request:**
+- HTTP method: POST.
+- Endpoint: `http://localhost:3000/tasks`
+- Request body: raw.
+- Request body conent type: `JSON`.
+- Request body:
+`[ {"taskDescription": "new task 1", "listIdReference": 1  },{"taskDescription": "new task 2", "listIdReference": 1  },{"taskDescription": "new task 3", "listIdReference": 1  }]`
+- Send the request.
+- You will get response likie this:
+ `{    "message": "Adding task or tasks done successfully",
```
Comment (roxana-lafuente) [inline Week3/homework/readme.md]: Great error message. We could improve it even more by adding the new ID of the item created. Something like for instance: "Adding task or task done successfully with id=20"

File: Week3/homework/api/task/taskController.js [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+    markCompleted: async function(req, res, next) {
+        const taskId = req.params.taskId;
+        const sql = `UPDATE adham_database_hw3.tasks SET task_completed = 1 WHERE task_id = ?`;
+        try {
+            let sqlResult = await queryPromise(sql, [taskId]);
+            res.status(200).json({
+                message: `Trying to mark the task with id ${taskId} as completed.`,
+                info: sqlResult.affectedRows === 0 ?
+                    'Task is NOt marked: that task is not existed.' :
```
Comment (roxana-lafuente) [inline Week3/homework/api/task/taskController.js]: Great for this!

File: Week3/homework/readme.md (hunk 2 above)
Comment (Adham-Elias-Botours) [reply Week3/homework/readme.md]: yes, I though about the same thing.. The result object is very helpful for that target.
I couldn't apply all my thoughts, I have ben already late to submit the homework.

**Comment by Adham-Elias-Botours** [comment]
> I'm super happy with your homework! Thanks for the super good work!
> 
> The only thing I would like you to add if possible is a way to see the TODO lists I have and the items in a specific TODO list.
> 
> **Edit**: For me this is a must-have functionality because besides adding and deleting items from the lists, the user will want to see what needs to be done. However, this can be considered as an improvement., therefore, I'm ok with not adding it. So I approve.
Dear @roxana-lafuente 
Thanks for your time on reviewing and approving this homework.

**Pull request closed without merging**